solr/core/src/test/org/apache/solr/handler/admin/V2CoresAPIMappingTest.java

```diff
@@ -0,0 +1,146 @@
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+import static org.apache.solr.common.params.CollectionAdminParams.NUM_SHARDS;
+import static org.apache.solr.common.params.CommonAdminParams.ASYNC;
+import static org.apache.solr.common.params.CommonParams.ACTION;
+import static org.apache.solr.common.params.CoreAdminParams.*;
+import static org.apache.solr.common.params.CoreAdminParams.CoreAdminAction.CREATE;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+
+/**
+ * Unit tests for the /cores APIs.
+ *
+ * Note that the V2 requests made by these tests are not necessarily semantically valid.  They shouldn't be taken as
+ * examples. In several instances, mutually exclusive JSON parameters are provided.  This is done to exercise conversion
+ * of all parameters, even if particular combinations are never expected in the same request.
+ */
+public class V2CoresAPIMappingTest extends SolrTestCaseJ4 {
+    private ApiBag apiBag;
+    private ArgumentCaptor<SolrQueryRequest> queryRequestCaptor;
+    private CoreAdminHandler mockCoreAdminHandler;
+
+    @Before
+    public void setUpMocks() {
+        mockCoreAdminHandler = mock(CoreAdminHandler.class);
+        queryRequestCaptor = ArgumentCaptor.forClass(SolrQueryRequest.class);
+
+        apiBag = new ApiBag(false);
+        apiBag.registerObject(new CreateCoreAPI(mockCoreAdminHandler));
+    }
+
+    @Test
+    public void testCreateCoreAllProperties() throws Exception {
+        final SolrParams v1Params = captureConvertedV1Params("/cores", "POST",
+                "{'create': {" +
+                        "'name': 'someCoreName', " +
+                        "'instanceDir': 'someInstanceDir', " +
+                        "'dataDir': 'someDataDir', " +
+                        "'ulogDir': 'someUpdateLogDirectory', " +
+                        "'schema': 'some-schema-file-name', " +
+                        "'config': 'some-config-file-name', " +
+                        "'configSet': 'someConfigSetName', " +
+                        "'loadOnStartup': true, " +
+                        "'isTransient': true, " +
+                        "'shard': 'someShardName', " +
+                        "'collection': 'someCollectionName', " +
+                        "'replicaType': 'TLOG', " +
+                        "'coreNodeName': 'someNodeName', " +
+                        "'numShards': 123, " +
+                        "'roles': ['role1', 'role2'], " +
+                        "'properties': {'prop1': 'val1', 'prop2': 'val2'}, " +
+                        "'newCollection': true, " +
+                        "'async': 'requestTrackingId' " +
+                        "}}");
+
+        // TODO Delete apispec file
```

Comment:
I'm sure you haven't forgotton about this.

Reply:
;-).  You did complete these two TODO's

Reply:
Haha, yeah, these are remnants from a TODO list I wrote myself at some point.  Will remove.